### PR TITLE
Add missing break in plugins list

### PR DESCRIPTION
**Pull request by jonahwilliams**
## Description

If plugins is not a map, the validation crashes. Ensure we break instead of continuing the validation

```
NoSuchMethodError: The method 'containsKey' was called on null.
Receiver: null
Tried calling: containsKey("platforms")

  | at Object.noSuchMethod | (object_patch.dart:51 )
-- | -- | --
  | at Plugin.validatePluginYaml | (plugins.dart:153 )
  | at _validateFlutter | (flutter_manifest.dart:407 )
  | at _validate | (flutter_manifest.dart:343 )
  | at FlutterManifest._createFromYaml | (flutter_manifest.dart:45 )
  | at FlutterManifest.createFromString | (flutter_manifest.dart:40 )
  | at FlutterManifest.createFromPath | (flutter_manifest.dart:34 )
  | at FlutterProject._readManifest | (project.dart:186 )
  | at FlutterProjectFactory.fromDirectory.<anonymous closure> | (project.dart:38 )
  | at _LinkedHashMapMixin.putIfAbsent | (compact_hash.dart:291 )
  | at FlutterProjectFactory.fromDirectory | (project.dart:37 )
  | at FlutterProject.fromDirectory | (project.dart:69 )
  | at FlutterProject.current | (project.dart:73 )
  | at FlutterCommand.findAllTargetDevices | (flutter_command.dart:582 )
  | at <asynchronous gap> | (async )
  | at RunCommand.validateCommand | (run.dart:274 )
  | at <asynchronous gap> | (async )
  | at FlutterCommand.verifyThenRunCommand | (flutter_command.dart:532 )
  | at <asynchronous gap> | (async )
  | at FlutterCommand.run.<anonymous closure> | (flutter_command.dart:457 )
  | at <asynchronous gap> | (async )
  | at AppContext.run.<anonymous closure> | (context.dart:157 )
  | at <asynchronous gap> | (async )
  | at _rootRun | (zone.dart:1124 )
  | at _CustomZone.run | (zone.dart:1021 )
  | at _runZoned | (zone.dart:1516 )
  | at runZoned | (zone.dart:1463 )
  | at AppContext.run | (context.dart:156 )
  | at <asynchronous gap> | (async )
  | at FlutterCommand.run | (flutter_command.dart:446 )
  | at CommandRunner.runCommand | (command_runner.dart:197 )
  | at <asynchronous gap> | (async )
  | at FlutterCommandRunner.runCommand.<anonymous closure> | (flutter_command_runner.dart:416 )
  | at <asynchronous gap> | (async )
  | at AppContext.run.<anonymous closure> | (context.dart:157 )
  | at <asynchronous gap> | (async )
  | at _rootRun | (zone.dart:1124 )
  | at _CustomZone.run | (zone.dart:1021 )
  | at _runZoned | (zone.dart:1516 )
  | at runZoned | (zone.dart:1463 )
  | at AppContext.run | (context.dart:156 )
  | at <asynchronous gap> | (async )
  | at FlutterCommandRunner.runCommand | (flutter_command_runner.dart:367 )
  | at <asynchronous gap> | (async )
  | at CommandRunner.run.<anonymous closure> | (command_runner.dart:112 )
  | at new Future.sync | (future.dart:224 )
  | at CommandRunner.run | (command_runner.dart:112 )
  | at FlutterCommandRunner.run | (flutter_command_runner.dart:251 )
  | at run.<anonymous closure>.<anonymous closure> | (runner.dart:63 )
  | at <asynchronous gap> | (async )
  | at _rootRun | (zone.dart:1124 )
  | at _CustomZone.run | (zone.dart:1021 )
  | at _runZoned | (zone.dart:1516 )
  | at runZoned | (zone.dart:1500 )
  | at run.<anonymous closure> | (runner.dart:61 )
  | at <asynchronous gap> | (async )
  | at AppContext.run.<anonymous closure> | (context.dart:157 )
  | at <asynchronous gap> | (async )
  | at _rootRun | (zone.dart:1124 )
  | at _CustomZone.run | (zone.dart:1021 )
  | at _runZoned | (zone.dart:1516 )
  | at runZoned | (zone.dart:1463 )
  | at AppContext.run | (context.dart:156 )
  | at <asynchronous gap> | (async )
  | at runInContext | (context_runner.dart:63 )
  | at <asynchronous gap> | (async )
  | at run | (runner.dart:50 )
  | at main | (executable.dart:65 )
  | at <asynchronous gap> | (async )
  | at main | (flutter_tools.dart:8 )
  | at _startIsolate.<anonymous closure> | (isolate_patch.dart:303 )
  | at _RawReceivePortImpl._handleMessage | (isolate_patch.dart:172 )

```

### CODE REVIEW

**Review by christopherfujino — APPROVED** [review]
lgtm

**Comment by christopherfujino** [comment]
still lgtm

**Review by iskakaushik — APPROVED** [review]
LGTM

File: dev/devicelab/lib/framework/utils.dart [hunk 1]
```diff
@@ -314,6 +314,13 @@ Future<int> exec(
       .transform<String>(const LineSplitter())
       .listen((String line) {
         print('stderr: $line');
+        // grab gradle daemon log file.
+        if (line.contains('log file:')) {
+          final File file = File(line.split(':').last.trim());
```
Comment (jonahwilliams) [inline dev/devicelab/lib/framework/utils.dart]: If gradle crashes it writes a log file, this ensures that it shows up in our crash logs

**Comment by jonahwilliams** [comment]
For readers at home, the stack trace is:

```
[module_host_with_custom_build_test] [STDOUT] crash: 
[module_host_with_custom_build_test] [STDOUT] crash: ```
[module_host_with_custom_build_test] [STDOUT] crash: NoSuchMethodError: The method 'containsKey' was called on null.
[module_host_with_custom_build_test] [STDOUT] crash: Receiver: null
[module_host_with_custom_build_test] [STDOUT] crash: Tried calling: containsKey("platforms")
[module_host_with_custom_build_test] [STDOUT] crash: 
[module_host_with_custom_build_test] [STDOUT] crash: ?? | at??Object.noSuchMethod | (object_patch.dart:51 )
[module_host_with_custom_build_test] [STDOUT] crash: -- | -- | --
[module_host_with_custom_build_test] [STDOUT] crash: ?? | at??Plugin.validatePluginYaml | (plugins.dart:153 )
[module_host_with_custom_build_test] [STDOUT] crash: ?? | at??_validateFlutter | (flutter_manifest.dart:407 )
[module_host_with_custom_build_test] [STDOUT] crash: ?? | at??_validate | (flutter_manifest.dart:343 )
[module_host_with_custom_build_test] [STDOUT] crash: ?? | at??FlutterManifest._createFromYaml | (flutter_manifest.dart:45 )
[module_host_with_custom_build_test] [STDOUT] crash: ?? | at??FlutterManifest.createFromString | (flutter_manifest.dart:40 )
[module_host_with_custom_build_test] [STDOUT] crash: ?? | at??FlutterManifest.createFromPath | (flutter_manifest.dart:34 )
[module_host_with_custom_build_test] [STDOUT] crash: ?? | at??FlutterProject._readManifest | (project.dart:186 )
[module_host_with_custom_build_test] [STDOUT] crash: ?? | at??FlutterProjectFactory.fromDirectory.<anonymous closure> | (project.dart:38 )
[module_host_with_custom_build_test] [STDOUT] crash: ?? | at??_LinkedHashMapMixin.putIfAbsent | (compact_hash.dart:291 )
[module_host_with_custom_build_test] [STDOUT] crash: ?? | at??FlutterProjectFactory.fromDirectory | (project.dart:37 )
[module_host_with_custom_build_test] [STDOUT] crash: ?? | at??FlutterProject.fromDirectory | (project.dart:69 )
[module_host_with_custom_build_test] [STDOUT] crash: ?? | at??FlutterProject.current | (project.dart:73 )
[module_host_with_custom_build_test] [STDOUT] crash: ?? | at??FlutterCommand.findAllTargetDevices | (flutter_command.dart:582 )
[module_host_with_custom_build_test] [STDOUT] crash: ?? | at??<asynchronous gap> | (async )
[module_host_with_custom_build_test] [STDOUT] crash: ?? | at??RunCommand.validateCommand | (run.dart:274 )
[module_host_with_custom_build_test] [STDOUT] crash: ?? | at??<asynchronous gap> | (async )
[module_host_with_custom_build_test] [STDOUT] crash: ?? | at??FlutterCommand.verifyThenRunCommand | (flutter_command.dart:532 )
[module_host_with_custom_build_test] [STDOUT] crash: ?? | at??<asynchronous gap> | (async )
[module_host_with_custom_build_test] [STDOUT] crash: ?? | at??FlutterCommand.run.<anonymous closure> | (flutter_command.dart:457 )
[module_host_with_custom_build_test] [STDOUT] crash: ?? | at??<asynchronous gap> | (async )
[module_host_with_custom_build_test] [STDOUT] crash: ?? | at??AppContext.run.<anonymous closure> | (context.dart:157 )
[module_host_with_custom_build_test] [STDOUT] crash: ?? | at??<asynchronous gap> | (async )
[module_host_with_custom_build_test] [STDOUT] crash: ?? | at??_rootRun | (zone.dart:1124 )
[module_host_with_custom_build_test] [STDOUT] crash: ?? | at??_CustomZone.run | (zone.dart:1021 )
[module_host_with_custom_build_test] [STDOUT] crash: ?? | at??_runZoned | (zone.dart:1516 )
[module_host_with_custom_build_test] [STDOUT] crash: ?? | at??runZoned | (zone.dart:1463 )
[module_host_with_custom_build_test] [STDOUT] crash: ?? | at??AppContext.run | (context.dart:156 )
[module_host_with_custom_build_test] [STDOUT] crash: ?? | at??<asynchronous gap> | (async )
[module_host_with_custom_build_test] [STDOUT] crash: ?? | at??FlutterCommand.run | (flutter_command.dart:446 )
[module_host_with_custom_build_test] [STDOUT] crash: ?? | at??CommandRunner.runCommand | (command_runner.dart:197 )
[module_host_with_custom_build_test] [STDOUT] crash: ?? | at??<asynchronous gap> | (async )
[module_host_with_custom_build_test] [STDOUT] crash: ?? | at??FlutterCommandRunner.runCommand.<anonymous closure> | (flutter_command_runner.dart:416 )
[module_host_with_custom_build_test] [STDOUT] crash: ?? | at??<asynchronous gap> | (async )
[module_host_with_custom_build_test] [STDOUT] crash: ?? | at??AppContext.run.<anonymous closure> | (context.dart:157 )
[module_host_with_custom_build_test] [STDOUT] crash: ?? | at??<asynchronous gap> | (async )
[module_host_with_custom_build_test] [STDOUT] crash: ?? | at??_rootRun | (zone.dart:1124 )
[module_host_with_custom_build_test] [STDOUT] crash: ?? | at??_CustomZone.run | (zone.dart:1021 )
[module_host_with_custom_build_test] [STDOUT] crash: ?? | at??_runZoned | (zone.dart:1516 )
[module_host_with_custom_build_test] [STDOUT] crash: ?? | at??runZoned | (zone.dart:1463 )
[module_host_with_custom_build_test] [STDOUT] crash: ?? | at??AppContext.run | (context.dart:156 )
[module_host_with_custom_build_test] [STDOUT] crash: ?? | at??<asynchronous gap> | (async )
[module_host_with_custom_build_test] [STDOUT] crash: ?? | at??FlutterCommandRunner.runCommand | (flutter_command_runner.dart:367 )
[module_host_with_custom_build_test] [STDOUT] crash: ?? | at??<asynchronous gap> | (async )
[module_host_with_custom_build_test] [STDOUT] crash: ?? | at??CommandRunner.run.<anonymous closure> | (command_runner.dart:112 )
[module_host_with_custom_build_test] [STDOUT] crash: ?? | at??new Future.sync | (future.dart:224 )
[module_host_with_custom_build_test] [STDOUT] crash: ?? | at??CommandRunner.run | (command_runner.dart:112 )
[module_host_with_custom_build_test] [STDOUT] crash: ?? | at??FlutterCommandRunner.run | (flutter_command_runner.dart:251 )
[module_host_with_custom_build_test] [STDOUT] crash: ?? | at??run.<anonymous closure>.<anonymous closure> | (runner.dart:63 )
[module_host_with_custom_build_test] [STDOUT] crash: ?? | at??<asynchronous gap> | (async )
[module_host_with_custom_build_test] [STDOUT] crash: ?? | at??_rootRun | (zone.dart:1124 )
[module_host_with_custom_build_test] [STDOUT] crash: ?? | at??_CustomZone.run | (zone.dart:1021 )
[module_host_with_custom_build_test] [STDOUT] crash: ?? | at??_runZoned | (zone.dart:1516 )
[module_host_with_custom_build_test] [STDOUT] crash: ?? | at??runZoned | (zone.dart:1500 )
[module_host_with_custom_build_test] [STDOUT] crash: ?? | at??run.<anonymous closure> | (runner.dart:61 )
[module_host_with_custom_build_test] [STDOUT] crash: ?? | at??<asynchronous gap> | (async )
[module_host_with_custom_build_test] [STDOUT] crash: ?? | at??AppContext.run.<anonymous closure> | (context.dart:157 )
[module_host_with_custom_build_test] [STDOUT] crash: ?? | at??<asynchronous gap> | (async )
[module_host_with_custom_build_test] [STDOUT] crash: ?? | at??_rootRun | (zone.dart:1124 )
[module_host_with_custom_build_test] [STDOUT] crash: ?? | at??_CustomZone.run | (zone.dart:1021 )
[module_host_with_custom_build_test] [STDOUT] crash: ?? | at??_runZoned | (zone.dart:1516 )
[module_host_with_custom_build_test] [STDOUT] crash: ?? | at??runZoned | (zone.dart:1463 )
[module_host_with_custom_build_test] [STDOUT] crash: ?? | at??AppContext.run | (context.dart:156 )
[module_host_with_custom_build_test] [STDOUT] crash: ?? | at??<asynchronous gap> | (async )
[module_host_with_custom_build_test] [STDOUT] crash: ?? | at??runInContext | (context_runner.dart:63 )
[module_host_with_custom_build_test] [STDOUT] crash: ?? | at??<asynchronous gap> | (async )
[module_host_with_custom_build_test] [STDOUT] crash: ?? | at??run | (runner.dart:50 )
[module_host_with_custom_build_test] [STDOUT] crash: ?? | at??main | (executable.dart:65 )
[module_host_with_custom_build_test] [STDOUT] crash: ?? | at??<asynchronous gap> | (async )
[module_host_with_custom_build_test] [STDOUT] crash: ?? | at??main | (flutter_tools.dart:8 )
[module_host_with_custom_build_test] [STDOUT] crash: ?? | at??_startIsolate.<anonymous closure> | (isolate_patch.dart:303 )
[module_host_with_custom_build_test] [STDOUT] crash: ?? | at??_RawReceivePortImpl._handleMessage | (isolate_patch.dart:172 )
```

**Comment by jonahwilliams** [comment]
This is claiming to that the validation method is on line 407 - however this was only true before I added the `break`. It looks like the gradle plugin tests are using old code?

File: dev/bots/test.dart [hunk 2]
```diff
@@ -948,6 +949,19 @@ String get androidSdkRoot {
...
+    File(flutterToolStamp).deleteSync();
+  } on FileSystemException {
+    // Do nothing.
+  }
+  await runCommand(flutter,
+    <String>['doctor'],
+    workingDirectory: flutterRoot,
+    expectNonZeroExit: false,
+    outputMode: OutputMode.discard,
+  );
```
Comment (Hixie) [inline dev/bots/test.dart]: there is code elsewhere in this file to do the same thing. please refactor the file to have just one place that does this (called from wherever it's needed)

File: dev/devicelab/lib/framework/utils.dart (hunk 1 above)
Comment (Hixie) [reply dev/devicelab/lib/framework/utils.dart]: good call

**Comment by jonahwilliams** [comment]
Closing because I can't figure this out

**Pull request closed without merging**